packages/test/end-to-end-tests/src/test/agentScheduler.spec.ts

```diff
@@ -105,8 +107,9 @@ const tests = (argsFactory: () => ITestObjectProvider) => {
             // Set a key in the root map. The Container is created in "read" mode and so it cannot currently pick
             // tasks. Sending an op will switch it to "write" mode.
             dataObject1._root.set("tempKey1", "tempValue1");
-            await args.opProcessingController.process();
-
+            while (!container1.deltaManager.active) {
```

Comment:
>while (!container1.deltaManager.active) [](start = 12, length = 39)

just wondering, how come this is needed in some places and not others?

Reply:
i'm a bit confused by it as well. i'd like to see #4813 to get rid of all these hacks